tests/python/relax/test_transform_fuse_ops_by_pattern.py

```diff
@@ -460,5 +496,10 @@ def test_multiple_calls_same_extern():
     check(Conv2dx2, [("cutlass.conv2d", pat)], Conv2dx2_partitioned, annoatate_codegen=True)
 
```

Comment:
Since we introduced new variable `dedup_params`, can we add a test for it?

Reply:
That's an internal implementation detail that is already exercised by all tests.

Reply:
If so, can we add comment which case would enable/disable this param? It is always good to have some high-level info :)

Reply:
So the purpose of using `dedup_params = false` is to decouple "duplication check" and "param update" (the function is literally called `CheckDefAndUpdateParams`). The only place where the function is called with `dedup_params = true` is https://github.com/apache/tvm/blob/46b500cb5eebfd2b75d7df97c009463cf4a75ca6/src/relax/transform/fuse_ops.cc#L402. I just found that I can pass all tests after removing this call.

So what we can do is to remove `dedup_params`, and rename `CheckDefAndUpdateParams` to just `UpdateParams`, so that param deduplication becomes the responsibility of the caller. I'll make such change.

Reply:
Okay removed `dedup_params` and decoupled deduplication and param update in `CheckDefAndUpdateParam`.

Reply:
Sounds good to me. Thank you!